### PR TITLE
feat: add formatToOctal function and documentation

### LINKED ISSUE #126: Formatting: Add formatToOctal function
**Description:**
The formatToOctal(num) function will convert a given decimal number into its octal (base-8) representation.

**Acceptance Criteria:**
```
8 → "10"

10 → "12"

255 → "377"

0 → "0"
```
Negative numbers should include a - prefix.

The function should support an optional { prefix: true } parameter to prepend "0o".
Example: formatToOctal(255, { prefix: true }) → "0o377".

Input validation for non-numeric inputs.

Add comprehensive unit tests for edge cases like 0, large numbers, and prefix handling.

### PR DESCRIPTION
**Pull request by ftonato**
Closes #126

### CODE REVIEW

File: src/formatting/index.ts [hunk 1]
```diff
@@ -33,5 +35,6 @@ export const formatting = {
   formatFileSize,
   formatOrdinal,
   formatList,
+  formatToOctal
   formatTemperature
```
[the remark this answers is not in the document]
Comment (ftonato) [reply src/formatting/index.ts]: I’m sorry — I had to resolve some conflicts and accidentally broke this one. Fixing it right now.
